lib/internal/Magento/Framework/HTTP/Client/Curl.php

```diff
@@ -239,13 +250,109 @@ public function get($uri)
      * @param array|string $params
      * @return void
      *
-     * @see \Magento\Framework\HTTP\Client#post($uri, $params)
+     * @see \Magento\Framework\HTTP\Client::post($uri, $params)
+     * @url https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.3
      */
     public function post($uri, $params)
     {
         $this->makeRequest("POST", $uri, $params);
     }
 
+    /**
+     * Make PUT request
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     *
+     * @url https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.4
+     */
+    public function put(string $uri, string|array $params): void
+    {
+        $this->makeRequest("PUT", $uri, $params);
+    }
+
+    /**
+     * Make DELETE request
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     *
+     * @url https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.5
+     */
+    public function delete($uri)
+    {
+        $this->makeRequest("DELETE", $uri, $params);
```

Comment:
@lbajsarowicz there's a bug in here - it uses $params for makeRequest but is not receiving any in the method head, there's only $uri there.

Reply:
Nice catch. I'm handling it.